### PR TITLE
Updated gitignore file to include IntelliJ related files

### PR DESCRIPTION
**Pull request by TheItachiUchiha**
Also includes ignoring javadoc dir

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,18 @@
...
+.idea/
+*.iml
+
+# Eclipse related
+.settings/
+.classpath
+.project
+
+# Miscellenous
+*.properties
```
Comment (mangstadt) [inline .gitignore]: I think the individual .properties files should be specified here instead of all .properties files.

Comment (TheItachiUchiha) [inline .gitignore]: Sure, agree. Unfortunately, I will be able to update this PR in 3 days time :(

Comment (mangstadt) [inline .gitignore]: Dang it man!  What am I paying you for? xD

File: .gitignore [hunk 2]
```diff
@@ -1,5 +1,18 @@
...
+*.log
```
Comment (TheItachiUchiha) [inline .gitignore]: @mangstadt Do you think *.log is ok?

Comment (mangstadt) [inline .gitignore]: I think it makes sense to leave it in, because the program will generate .log files if you run it from your development environment.

**Pull request merged**